src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,37 @@
+package duke.command;
+
+import duke.data.task.TaskList;
+import duke.message.Messages;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+public class FindCommand extends Command {
+
+    private final String keyword;
+
+    /**
+     * Initialises a new instance.
+     *
+     * @param keyword The keyword for finding
+     */
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    /**
+     * Search for the keyword within the task list and print a list of matching tasks
+     *
+     * @param tasks   The list of tasks known
+     * @param storage The storage that is used
+     * @param ui The ui that is used
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage)  {
+        TaskList machingTasks = tasks.search(this.keyword);
+        if (machingTasks.size() == 0) {
```

Comment:
Seems like there is a spelling mistake in the variable name (should be matchingTasks). 
Also instead of comparing the size to 0, you can do something like:
```
if (!matchingTasks.size()) {
}
```

Reply:
Thank you for your correction!! 
As for if (!matchingTasks.size()) , agree with this, however, the compiler will report error. In this case '==0' is easier to understand?

Reply:
ok ya sure